### PR TITLE
Week1 xiaoxue shen

### PR DESCRIPTION
**Pull request by shirleyshen06**

### CODE REVIEW

File: springboot-user(1)/src/main/java/com/chuwa/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    private UserRepo userRepo;
+
+    @Autowired
+    public UserServiceImpl(UserRepo userRepo) {
+        super();
+        this.userRepo = userRepo;
+    }
+
+    @Override
+    public User addUser(User user) {
```
Comment (TAIsRich) [inline springboot-user(1)/src/main/java/com/chuwa/service/impl/UserServiceImpl.java:25]: 请添加user的验证，然后再save，
user == null?
user 没有password？
if ( user  == null || user.getPassword == null) {
throw exception();
}

**Review by TAIsRich — CHANGES_REQUESTED** [review]

File: springboot-user(2)/src/main/java/com/chuwa/controller/userController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        User user = userServiceImpl.findUserByEmail(email);
+        return  new ResponseEntity<>(user,HttpStatus.OK);
+    }
+    @PutMapping("/update")
+    public ResponseEntity<User> updateEmployee(@RequestBody User user){
+        User updateUser = userServiceImpl.updateUser(user);
+        return new ResponseEntity<>(updateUser,HttpStatus.OK);
+    }
+
+    @DeleteMapping("delete/{id}")
```
Comment (TAIsRich) [inline springboot-user(2)/src/main/java/com/chuwa/controller/userController.java:42]: api命名有问题。
/users/{id}

File: springboot-user(2)/src/main/java/com/chuwa/controller/userController.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        return  new ResponseEntity<>(newUser, HttpStatus.CREATED);
+    }
+
+
+    @GetMapping("/find/{email}")
+    public ResponseEntity<User> getUserByEmail(@PathVariable("email") String email){
+        User user = userServiceImpl.findUserByEmail(email);
+        return  new ResponseEntity<>(user,HttpStatus.OK);
+    }
+    @PutMapping("/update")
```
Comment (TAIsRich) [inline springboot-user(2)/src/main/java/com/chuwa/controller/userController.java:36]: /users/{id}

File: springboot-user(2)/src/main/java/com/chuwa/controller/userController.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+    @PostMapping("/signup")
+    public ResponseEntity<User> registerUser(@RequestBody User user){
+        User newUser = userServiceImpl.addUser(user);
+        return  new ResponseEntity<>(newUser, HttpStatus.CREATED);
+    }
+
+
+    @GetMapping("/find/{email}")
```
Comment (TAIsRich) [inline springboot-user(2)/src/main/java/com/chuwa/controller/userController.java:31]: /users/{email}

**Review by TAIsRich — CHANGES_REQUESTED** [review]

File: springboot-user(2)/src/main/java/com/chuwa/controller/userController.java (hunk 4 above)
Comment (shirleyshen06) [reply springboot-user(2)/src/main/java/com/chuwa/controller/userController.java:31]: Screen Shot 2022-05-03 at 3 46 26 PM
@RequestMapping("/user").   at top ,why I have to add "user" again ?

**Pull request merged**